agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -1393,6 +1393,10 @@ func (dg *dockerGoClient) Stats(ctx context.Context, id string, inactivityTimeou
 			stream := true
 			resp, err = client.ContainerStats(subCtx, id, stream)
 			if err != nil {
+				if errors.Is(err, context.Canceled) {
```

Comment:
I don't think the client is the right place to ignore this error. 
Can we still return it and ignore it where this method is invoked instead?

Reply:
i guess we can have the client wrapping this error with `%w`....and then we could call this at a higher level, I will try it

Reply:
That works. Alternatively you could just use the plain error when it's  `context.Canceled`
```golang
if errors.Is(err, context.Canceled) {
    errC <- err
    return
}
```